### PR TITLE
travis multiple environment deployment

### PR DESCRIPTION
**Pull request by wiktorschmidt**
a little hack that allows to publish all platform.io environments to github

also adding 2 extra environments to platform.io: minimal & 8 x DS18x20

### CODE REVIEW

**Comment by wiktorschmidt** [comment]
It's failing in the minimal environment for some reason:
```
.pioenvs/sonoff-minimal/src/sonoff.ino.o:(.text._Z9GPIO_initv+0x2c): undefined reference to `dht_setup(unsigned char, unsigned char)'
.pioenvs/sonoff-minimal/src/sonoff.ino.o: In function `GPIO_init()':
sonoff.ino.cpp:(.text._Z9GPIO_initv+0x11b): undefined reference to `dht_setup(unsigned char, unsigned char)'
```

**Comment by wiktorschmidt** [comment]
@arendst would love to get your feedback

File: platformio.ini [hunk 1]
```diff
@@ -10,6 +10,15 @@
...
 src_dir = sonoff
 
+; Sonoff minimal (ESP8266 based)
+[env:sonoff-minimal] ; Placeholder to be configured
+platform = espressif8266
+framework = arduino
+board = esp01_1m
+board_flash_mode = qio
+build_flags = -Wl,-Tesp8266.flash.1m0.ld -DMQTT_MAX_PACKET_SIZE=512 -DBE_MINIMAL
+lib_deps = PubSubClient, NeoPixelBus, IRremoteESP8266, ArduinoJSON
```
Comment (wiktorschmidt) [inline platformio.ini]: not sure if all those libs are needed in the minimal env

**Comment by wiktorschmidt** [comment]
Example release with all 4 environments: https://github.com/wiktorschmidt/Sonoff-Tasmota/releases/tag/v5.1.3.envs

**Comment by arendst** [comment]
I'm currently OoO. Will be back next week and chck.

**Comment by arendst** [comment]
Very nice! I will board this boat.

**Comment by wiktorschmidt** [comment]
🎉

**Pull request closed without merging**